engine/src/main/java/com/redhat/cloud/notifications/processors/webhooks/WebhookTypeProcessor.java

```diff
@@ -202,6 +202,16 @@ public void doHttpRequest(Event event, Endpoint endpoint, HttpRequest<Buffer> re
 
                 boolean serverError = false;
                 boolean shouldResetEndpointServerErrors = false;
+                if (isEmailEndpoint) {
+                    JsonObject details = new JsonObject();
+                    try {
+                        int totalRecipients = payload.getJsonArray("emails").getJsonObject(0).getJsonArray("bccList").size();
+                        details.put("total_recipients", totalRecipients);
+                        history.setDetails(details.getMap());
+                    } catch (Exception ex) {
+                        Log.info("This email payload doesn't use bccList");
+                    }
+                }
```

Comment:
`history#details` will be overwritten in case of exception or `INTERNAL_FAILURE` status. I think this block should move a few lines lower, inside the `if (resp.statusCode() >= 200 && resp.statusCode() < 300)` block.

Reply:
You are right, I used a slite different approach to preserve it in case of INTERNAL_FAILURE